src/Nethermind/Nethermind.Trie/Pruning/TinyTreePath.cs

```diff
@@ -43,7 +44,8 @@ public TreePath ToTreePath()
     public bool Equals(TinyTreePath other) => _data == other._data;
     public bool Equals(in TinyTreePath other) => _data == other._data;
     public override bool Equals(object? obj) => obj is TinyTreePath other && Equals(other);
-    public override int GetHashCode() => _data.GetHashCode();
+    // Need more variance than straightforward long.GetHashCode() as it determines lock contention in ConcurrentDictionary
+    public override int GetHashCode() => (int)BitOperations.Crc32C((uint)_data, ((ulong)_data & 0xffff_ffff_0000_0000) | ~(uint)_data);
```

Comment:
Reduced the contention in a couple ways, though this one is of probably most interesting to @asdacap